sdks/python/apache_beam/runners/worker/bundle_processor.py

```diff
@@ -1257,7 +1257,18 @@ def monitoring_infos(self):
       all_monitoring_infos_dict.update(
           op.monitoring_infos(transform_id, dict(tag_to_pcollection_id)))
 
-    return list(all_monitoring_infos_dict.values())
+    # Filter distributions that have values of zero counts.
+    filtered_dict = {}
```

Comment:
A few comments

1) The monitoring_info collection is a performance-sensitive part of Beam, so I am vary of adding additional processing here
2) This change will affect only new SDKs, runners would still have to accommodate the behavior for older SDKs.

3) Is it possible to not emit the empty distributions when they are initially created, so that they don't appear in all_monitoring_infos_dict in the first place?

Reply:
Done, did it so if the distribution creation returns an error, it never is added to all_monitoring_infos_dict